hudi-client/hudi-client-common/src/main/java/org/apache/hudi/table/action/clean/CleanPlanner.java

```diff
@@ -180,6 +184,11 @@ private List<String> getPartitionPathsForCleanByCommits(Option<HoodieInstant> in
     return getPartitionPathsForFullCleaning();
   }
 
+  private boolean isAfterCommitsTimelineStarts(HoodieActiveTimeline timeline, String instant) {
+    return timeline.getCommitsTimeline().firstInstant().isPresent()
+        && compareTimestamps(instant, GREATER_THAN_OR_EQUALS, timeline.getCommitsTimeline().firstInstant().get().getTimestamp());
```

Comment:
The commits timeline can be reused and you can still utilize the interface `isBeforeTimelineStarts`